src/renderer/editor-mosaic.ts

```diff
@@ -352,7 +355,29 @@ export class EditorMosaic {
   }
 
   public mainEntryPointFile(): EditorId | undefined {
-    return Array.from(this.files.keys()).find((id) => isMainEntryPoint(id));
+    if (!this.mainEntryPoint || !this.files.get(this.mainEntryPoint)) {
+      const entryPoint = Array.from(this.files.keys()).find((id) =>
+        isMainEntryPoint(id),
+      );
+      if (entryPoint) this.mainEntryPoint = entryPoint;
+      return entryPoint;
+    }
+    return this.mainEntryPoint;
```

Comment:
```suggestion
  public mainEntryPointFile(): EditorId | null {
    if (this.mainEntryPoint && this.files.get(this.mainEntryPoint)) {
      return this.mainEntryPoint;
    }

    for (const [id] of this.files) {
      if (isMainEntryPoint(id)) {
        this.mainEntryPoint = id;
        break;
      }
    }

    return this.mainEntryPoint;
```

Make the code more intuitive and optimize performance a bit.